docs/DeveloperGuide.md

```diff
@@ -245,6 +245,56 @@ Due to the repercussions of Alternative 2 and the efficiency of Alternative 1, w
 
 ### 3.3 Show feature
 
+The show feature enables users to show tasks by specifying the given week or date.
+
+#### 3.3.1 Implementation
+
+##### ShowCommand class
+
+The `ShowCommand` class extends the `Command` class. It manages the showing of tasks specified by the user based on the week number or date provided. It contains a `String` representing its command word to be used by the parser, a `String` representing its usage to be displayed if used incorrectly, a `String` representing the successful deletion of a task, and a `List<Index>`, `targetIndexes`, which contains the indexes of all tasks to be deleted.
+
+The `execute` method in `DeleteCommand` overrides that in `Command`. In this implementation, it exemplifies defensive programming by ensuring the `model` provided is non-`null`. It then checks if the indexes provided by the user are valid for the current list shown, and continues only if they are all valid (between 1 and the total number of items in the task list). A `CommandException` is thrown in cases of invalid indexes. In the happy path, the tasks are deleted iteratively starting from the last index provided to the first, to prevent future deletions from operating on wrong indexes due to the task list updating itself in each iteration.
+
+##### ShowCommandParser class
+
+The `ShowCommandParser` class implements the `Parser<DeleteCommand>` interface. It manages the parsing of the arguments (index(es) in the case of a delete command) in the user input.
+The `parse` method in `ShowCommandParser` first converts the argument provided into a `List<Index>`. It then returns a `DeleteCommand` back to `UniFyParser`, initialized with the `List<Index>`.
+
+##### Usage Scenario
+
+The following demonstrates a usage scenario where the user wants to delete the first, second and third item in her/his task list.
+
+1. The method `execute("delete 1 2 3")` inside LogicManager calls the `parseCommand` method of `UniFyParser`.
```

Comment:
I think the wrong command is used here. Do remember to edit the Usage and include the appropriate images.

Reply:
DG changes will be updated in the following push. Thank you for catching the error!